### PR TITLE
Refactor flaky local scoring tests

### PR DESCRIPTION
**Pull request by Jauntbox**
**Related issues**
N/A

**Describe the proposed solution**
The current local scoring tests are flaky when xgboost models are included because the dataset used is a tiny 8-row hardcoded dataset. This can cause the train/validation splits to often contain all of a single class which causes xgboost models to throw an error.

This PR makes the dataset used a synthetic dataset of adjustable size (now 100 rows) to fix this problem.

**Describe alternatives you've considered**
We could also have used the full hardcoded Titanic dataset, but this was much easier for me.

**Additional context**
N/A

### CODE REVIEW

**Comment by Jauntbox** [comment]
Oof, didn't realize you weren't a "code owner", I guess @leahmcguire needs to approve this too

File: local/src/test/scala/com/salesforce/op/local/OpWorkflowModelLocalTest.scala [hunk 1]
```diff
@@ -57,14 +56,31 @@ import org.slf4j.LoggerFactory
 
 
 @RunWith(classOf[JUnitRunner])
-class OpWorkflowModelLocalTest extends FlatSpec with PassengerSparkFixtureTest with TestCommon {
-
+class OpWorkflowModelLocalTest extends FlatSpec with TestSparkContext with TempDirectoryTest with TestCommon {
```
Comment (tovbinm) [inline local/src/test/scala/com/salesforce/op/local/OpWorkflowModelLocalTest.scala:59]: Where do you use temp directory here?

Comment (Jauntbox) [reply local/src/test/scala/com/salesforce/op/local/OpWorkflowModelLocalTest.scala:59]: For example, here: https://github.com/salesforce/TransmogrifAI/blob/30cd461968ab9df9648afc7e6eb7f31a9d8f155f/local/src/test/scala/com/salesforce/op/local/OpWorkflowModelLocalTest.scala#L207

The test originally pulled it in via `PassengerSparkFixtureTest`, but I only needed things in `TempDirectoryTest`.

Comment (tovbinm) [reply local/src/test/scala/com/salesforce/op/local/OpWorkflowModelLocalTest.scala:59]: Got it.

**Pull request merged**